preview-src/pullRequestOverviewRenderer.ts

```diff
@@ -227,57 +227,104 @@ function groupBy<T>(arr: T[], fn: (el: T) => string): { [key: string]: T[] } {
 	}, Object.create(null));
 }
 
-export function renderComment(comment: CommentEvent | Comment): string {
-	return `<div class="comment-container" data-type="comment">
-
-	<div class="review-comment" role="treeitem">
-		<div class="review-comment-contents comment">
-			<div class="avatar-container">
-				<a class="avatar-link" href="${comment.user.html_url}"><img class="avatar" src="${comment.user.avatar_url}"></a>
-			</div>
-			<div class="review-comment-container">
-				<div class="review-comment-header">
-					<a class="author" href="${comment.user.html_url}">${comment.user.login}</a>
-					<div class="timestamp">${moment(comment.created_at).fromNow()}</div>
-				</div>
-				<div class="comment-body">
-					${md.render(emoji.emojify(comment.body))}
-				</div>
-			</div>
-		</div>
-	</div>
-</div>`;
+function renderUserIcon(iconLink: string, iconSrc: string): HTMLElement {
+	const iconContainer = document.createElement('div');
+	iconContainer.className = 'avatar-container';
+
+	const avatarLink = document.createElement('a');
+	avatarLink.className = 'avatar-link';
+	(<HTMLAnchorElement>avatarLink).href = iconLink;
+
+	const avatar = document.createElement('img');
+	avatar.className = 'avatar';
+	(<HTMLImageElement>avatar).src = iconSrc;
+
+	iconContainer.appendChild(avatarLink).appendChild(avatar);
+
+	return iconContainer;
 }
 
-export function renderCommit(timelineEvent: CommitEvent): string {
+export function renderComment(comment: CommentEvent | Comment, additionalClass?: string): HTMLElement {
+	const commentContainer = document.createElement('div');
+	commentContainer.id = `comment${comment.id.toString()}`;
+	commentContainer.classList.add('comment-container', 'comment');
+
+	if (additionalClass) {
+		commentContainer.classList.add(additionalClass);
+	}
+
+	const userIcon = renderUserIcon(comment.user.html_url, comment.user.avatar_url);
+	const reviewCommentContainer = document.createElement('div');
+	reviewCommentContainer.className = 'review-comment-container';
+	commentContainer.appendChild(userIcon);
+	commentContainer.appendChild(reviewCommentContainer);
+
+	const commentHeader = document.createElement('div');
+	commentHeader.className = 'review-comment-header';
+	const authorLink = document.createElement('a');
```

Comment:
I don't know what our style is for these, but it somehow seems nicer to me to cast to the appropriate type ASAP.

```suggestion
	const authorLink = <HTMLAnchorElement>document.createElement('a');
```

(It's also a bummer to me that TypeScript can't handle this more gracefully.)

Reply:
Yeah good point! It's silly to keep casting them everywhere instead of just declaring the correct type up front.

changed all instances of `const <x> = document.createElement` to be explicitly typed